torch/distributed/algorithms/model_averaging/utils.py

```diff
@@ -1,13 +1,13 @@
 # flake8: noqa C101
 import itertools
-from typing import Iterator
+from typing import Iterator, List, Union
 
 import torch
 import torch.distributed as dist
-
+import types
 
 def average_parameters(
-    params: Iterator[torch.nn.Parameter], process_group: dist.ProcessGroup
+    params: Union[Iterator[torch.nn.Parameter], List[dict]] , process_group: dist.ProcessGroup, comm_memory_efficient: bool=False
```

Comment:
Please check my other comments in this file. I think the only change needed in this file is s/view/reshape.

Reply:
agree.
i will fix later

Reply:
If the above changes still cannot fix the case you reported, then as you suggested, you can try changing the data type of `params` from `Iterator[Parameter]` to `List[Parameter]`.

Let's only make necessary incremental changes.

Reply:
see new comment https://github.com/pytorch/pytorch/pull/74894#discussion_r842465678